app/controllers/admin/resources_controller.rb

```diff
@@ -48,5 +56,22 @@ def destroy
     def resource_params
       params.require(:resource).permit(:name, :description, :quantity, :used, :conference_id)
     end
+
+    def get_update_flag(used_increment_flag)
+      if used_increment_flag.present?
+        @resource.used = params[:used_increment_flag].to_i == 1 ? (@resource.used - 1) : (@resource.used + 1)
+        @resource.save
+      else
+        @resource.update_attributes(resource_params)
+      end
+    end
+
+    def get_successful_update_flash(used_increment_flag)
+      if used_increment_flag.to_i == 1
+        "One #{@resource.name} freed."
+      else
+        "One more #{@resource.name} used."
```

Comment:
Update to the resource can be via Edit page or Increment/Decrement via Index page. `get_update_flag` action distinguish between above two updates in order to use proper update logic for a different scenario. `get_successful_update_flash` action is just used to generate flash for later update case. The sole purpose of the separate action was to make the code more readable.